tests/workspace-files.feature

```diff
@@ -0,0 +1,36 @@
+# This file is part of REANA.
+# Copyright (C) 2023 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+Feature: Workspace files
+
+    As a researcher,
+    I want to make sure that the workspace produces expected files,
+    so that I can be sure that the workflow outputs are correct.
+
+    Scenario: The workspace contains the expected input files
+        When the workflow is finished
+        Then the workspace should include "code/gendata.C"
+        And the workspace should include "code/fitdata.C"
+
+    Scenario: The code used to generate the plot is correct
+        When the workflow is finished
+        Then the file "code/fitdata.C" should contain "RooWorkspace* w = (RooWorkspace*) f->Get("w")"
+
+    Scenario: The generation step creates the data file
+        When the workflow is finished
+        Then the workspace should contain "results/data.root"
+        And the size of the file "results/data.root" should be between 150KiB and 160KiB
+
+    Scenario: The workflow generates the final plot
+        When the workflow is finished
+        Then the workspace should contain "results/plot.png"
+        And the sha256 checksum of the file "results/plot.png" should be "54165023803ce58f962816c47fe597c5df2cf24c26fa4821bb5e96f7d1b68d0b"
+        And all the outputs should be included in the workspace
```

Comment:
Removed this one, since this feature is dedicated to plots only.  (We could actually separate it into another scenario ("all the outputs") later on.)